### PR TITLE
Fix for 557: Provide request headers in various payload events

### PR DESCRIPTION
**Pull request by bodawei**
This is a draft fix for https://github.com/hapijs/good/issues/557 . In particular, this demonstrates what I have in mind there.  Happy to discuss this in more detail.

### CODE REVIEW

**Comment by bodawei** [comment]
I noticed that some of the Travis CI build failed.   I don't get these same problems on my local system (notably, one of these seems to be missing some module (!), and others are about a "leak" of WebAssembly).  Any insights into what is up with this?

**Comment by arb** [comment]
Yes. An update to `lab` fixes the memory leak issue (it's not really a leak btw, just a global variable). Missing modules I'm not sure about... I haven't seen that locally or in any of my Travis runs. There are still failing tests because, surprise surprise, things changed in the stream API so it doesn't act the same in the 4 versions of node good is tested against.

**Comment by bodawei** [comment]
I looked closer and noticed that all the errors in this build are the same as the ones from https://github.com/hapijs/good/pull/553. Given that that is merely a .md file change, seems like it's not likely my change caused these.  So, this change still seems to go.

File: lib/utils.js [hunk 1]
```diff
@@ -143,6 +147,10 @@ class RequestLog {
         this.method = request.method;
         this.path = request.path;
         this.config = internals.extractConfig(request);
+
+        if (reqOptions.headers) {
+            this.headers = request.raw.req.headers;
```
Comment (arb) [inline lib/utils.js]: Just to be safe incase this isn't there, use `hoek.reach`

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "good",
   "description": "Server and process monitoring plugin",
-  "version": "7.2.0",
+  "version": "7.3.0",
```
Comment (arb) [inline package.json]: Undo.

**Pull request merged**